src/core/retargeting_engine/python/deviceio_source_nodes/message_channel_source.py

```
@@ -0,0 +1,102 @@
+# SPDX-FileCopyrightText: Copyright (c) 2026 NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""
+Message channel source node.
+
+Converts DeviceIO MessageChannelMessagesTrackedT wrapper data for graph use.
+"""
+
+from typing import Any, TYPE_CHECKING, Deque
```

Ruff (0.15.7)

[warning] `typing.Deque` is deprecated, use `collections.deque` instead

(UP035)